releases/0.6/upgrading/splinter-v0.6-from-v0.4.md

```diff
@@ -0,0 +1,342 @@
+the LMDB files can be found if it is not in the default location.
+
+```
+$ splinter state migrate --in lmdb --out /var/lib/splinter/splinter_state.db
+```
+
+## Scabbard has been updated to version 2
+
+Version 2 of the scabbard service has been introduced in Splinter 0.6. This new
+version of scabbard is not compatible with the old version; this means that a
+scabbard v2 service cannot work on the same circuit as a scabbard v1 service.
+
+Splinter 0.6 supports backwards compatibility for scabbard v1. When upgrading to
+Splinter 0.6, any existing scabbard services will continue to work as they did
+before. New circuits created using the scabbard circuit template provided by the
+Splinter CLI will create scabbard v2 services.
+
+If a Splinter 0.6 node and a Splinter 0.4 node need to create a new circuit with
+scabbard v1 services, the circuit will need to be proposed manually (without the
+  scabbard circuit template).
+
+## The Splinter REST API is now secure by default
+
+In 0.4, the Splinter REST API did not require authentication and authorization.
+
+Splinter 0.6 provides a secure REST API for managing circuits, submitting
+transactions, reading state, and a variety of other operations. Clients must
+authenticate and be granted permissions to use the various `splinterd`
+endpoints, which requires some setup. For more information on how to configure
+this see [Configuring REST API Authorization]({% link
+docs/0.6/howto/configuring_rest_api_authorization.md %}).
+
+If you were using Biome in 0.4, the only change required is to update the biome
+flag provided to `splinterd`. It has been changed from `--enable-biome` to
+`--enable-biome-credentials`
+
+```
+$ splinterd --enable-biome-credentials
+```
+
+## AdminProtocolVersion has increased
+
+In 0.4, the AdminProtocolVersion was set to 1.
+
+In 0.6, due to several changes in the circuit API the AdminProtocolVersion is
+now set to 2. If you have a client that is compatible with 0.4, make sure to set
+the SplinterProtocolVersion header to 2.
+
+To see the updated API see the [REST API
+reference](https://www.splinter.dev/docs/0.6/api/#tag/Admin-Service)
+
+## Default authorization type is now Challenge Authorization
+
+When Splinter nodes connect, they must go through a “handshake” to verify the
+identity of the other node.
+
+In 0.4, Trust Authorization is the only authorization algorithm implemented.
+Trust Authorization takes the identity provided from the node without further
+verification.
+
+In 0.6, the default authorization type has been updated to Challenge
+Authorization. Challenge Authorization requires a node’s ID to be tied to a
+public key/private key pair and a node must prove they have access to that key
+by signing a random nonce, providing the resulting signature and their public
+key.
+
+This requires that a `splinterd` has its own system key that will be used for
+identification. The key must be stored in the SPLINTER_CONFIG_DIR/keys/. If
```

Comment:
stored at? in the `SPLINTER_CONFIG_DIR/keys/` directory?

Reply:
fixed (went with directory)